.github/workflows/kernel_test.yml

```diff
@@ -83,3 +83,29 @@ jobs:
       - name: Regression Test (MicroVM)
         id: regression_test_linux
         run: make run AUTO_TEST=regression ENABLE_KVM=0 QEMU_MACHINE=microvm RELEASE_MODE=1
+
+  vsock-test:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Run Vsock Server on Host
+        id: host_vsock_server
+        run: |
+          sudo modprobe vhost_vsock
+          sudo apt-get install socat
+          echo "Run vsock server on host...."
+          socat -dd vsock-listen:1234 SYSTEM:"sleep 3s; echo 'Hello from host';sleep 2s;kill -INT $$" &
+      - name: Run Vsock Client and Server on Guest
+        id: guest_vsock_client_server
+        run: |
+          docker run  --privileged --network=host --device=/dev/kvm -v ./:/root/asterinas asterinas/asterinas:0.4.2 \
+          make run AUTO_TEST=vsock ENABLE_KVM=0 QEMU_MACHINE=microvm RELEASE_MODE=1 &
+      - name: Run Vsock Client on Host
+        id: host_vsock_client
+        run: |
+          sleep 5m
+          echo "Run vsock client on host...."
+          echo "Hello from host" | socat -dd - vsock-connect:3:4321
+
```

Comment:
I suggest moving the vsock-test to a new yml file. This test is rather different with other test in env building and test steps so they should be put in different files.

The file may be named as `vsock_interaction.yml`, or other names you think is better.

Further, we hope any operations are initiated by client inside Asterinas. For example, instead of `sleep 5m` in `host_vsock_client`, the client inside Asterinas sends a request for starting a vsock client in host. This can provide a flexible way for testing through vsock, since we will add more test (for example, virtio-net test) based on vsock